internal/getproviders/package_authentication.go

```diff
@@ -359,15 +378,50 @@ type signatureAuthentication struct {
 //
 // Any failure in the process of validating the signature will result in an
 // unauthenticated result.
-func NewSignatureAuthentication(document, signature []byte, keys []SigningKey) PackageAuthentication {
+func NewSignatureAuthentication(document, signature []byte, keys []SigningKey, source *tfaddr.Provider) PackageAuthentication {
 	return signatureAuthentication{
-		Document:  document,
-		Signature: signature,
-		Keys:      keys,
+		Document:       document,
+		Signature:      signature,
+		Keys:           keys,
+		ProviderSource: source,
+	}
+}
+
+func (s signatureAuthentication) shouldEnforceGPGValidation() (bool, error) {
+	// we should enforce validation for all provider sources that are not the default provider registry
+	if s.ProviderSource != nil && s.ProviderSource.Hostname != tfaddr.DefaultProviderRegistryHost {
+		return true, nil
 	}
+
+	// if we have been provided keys, we should enforce GPG validation
+	if len(s.Keys) > 0 {
+		return true, nil
+	}
+
+	// otherwise if the environment variable is set to true, we should enforce GPG validation
+	enforceEnvVar, exists := os.LookupEnv(enforceGPGValidationEnvName)
+	return exists && enforceEnvVar == "true", nil
 }
 
 func (s signatureAuthentication) AuthenticatePackage(location PackageLocation) (*PackageAuthenticationResult, error) {
+	shouldValidate, err := s.shouldEnforceGPGValidation()
+	if err != nil {
+		return nil, fmt.Errorf("error determining if GPG validation should be enforced for pacakage %s: %s", location.String(), err.Error())
+	}
+
+	if !shouldValidate {
+		// As this is a temporary measure, we will log a warning to the user making it very clear what is happening
+		// and why. This will be removed in a future release.
+		log.Printf("[WARN] Skipping GPG validation of provider package %s as no keys were provided by the registry. See https://github.com/opentffoundation/opentf/pull/309 for more information.", location)
+
+		// construct an empty keyID to indicate that we are not validating and return no errors
+		// this is to force a successful authentication
+		// TODO: discuss if this key should be hardcoded to a value such as "UNKNOWN"?
+		return &PackageAuthenticationResult{result: signingSkipped, KeyID: ""}, nil
```

Comment:
Hard coded value sounds more reasonable ("SKIPPED" perhaps?)

Reply:
I've given this some thought lately, and since I added the `signingSkipped` result, this keyID is never read. I think leaving it as an empty string and removing the TODO may be best here. WDYT?

Reply:
SGTM